prov/util/src/util_mr_cache.c

```diff
@@ -566,3 +566,67 @@ int ofi_open_mr_cache(uint32_t version, void *attr, size_t attr_len,
 	*fid = cache_fid;
 	return 0;
 }
+
+static ssize_t ofi_copy_mr_iov(struct ofi_mr **mr, const struct iovec *iov,
+		size_t iov_count, uint64_t iov_offset, void *buf,
+		size_t size, int dir)
+{
+	uint64_t done = 0, len;
+	uint64_t hmem_iface, hmem_device;
+	char *hmem_buf;
+	size_t i;
+	int ret;
+
+	for (i = 0; i < iov_count && size; i++) {
+		len = iov[i].iov_len;
+
+		if (iov_offset > len) {
+			iov_offset -= len;
+			continue;
+		}
+
+		hmem_buf = (char *)iov[i].iov_base + iov_offset;
+		len -= iov_offset;
+		iov_offset = 0;
+
+		len = MIN(len, size);
+		if (!len)
+			continue;
+
+		if (mr && mr[i]) {
+			hmem_iface = mr[i]->iface;
+			hmem_device = mr[i]->device;
+		} else {
+			hmem_iface = FI_HMEM_SYSTEM;
+		}
+		if (dir == OFI_COPY_BUF_TO_IOV)
+			ret = ofi_copy_to_hmem(hmem_iface, hmem_device, hmem_buf,
+						(char *)buf + done, len);
+		else
+			ret = ofi_copy_from_hmem(hmem_iface, hmem_device,
+						(char *)buf + done, hmem_buf,
+						len);
+		if (ret)
+			return ret;
+
+		size -= len;
+		done += len;
+	}
```

Comment:
This duplicates most of ofi_copy_hmem_iov_buf().  Can we isolate the common code?